lib/oli_web/live/workspaces/utils.ex

```diff
@@ -178,24 +170,50 @@ defmodule OliWeb.Workspace.Utils do
   end
 
   attr :sidebar_expanded, :boolean, default: true
-  attr :title, :string, default: nil
+  attr :resource_title, :string, default: ""
 
   def title(assigns) do
     ~H"""
-    <div class="w-full">
-      <%= if @sidebar_expanded do %>
-        <h2 class="truncate text-[14px] h-[24px] font-bold ml-5 dark:text-[#B8B4BF] text-[#353740] tracking-[-1%] leading-6 uppercase">
-          <%= @title %>
-        </h2>
-      <% else %>
-        <div class="w-9 m-auto border border-zinc-500"></div>
-      <% end %>
+    <div class={"#{if @sidebar_expanded, do: "block", else: "hidden"} truncate text-[14px] h-[24px] font-bold ml-5 dark:text-[#B8B4BF] text-[#353740] tracking-[-1%] leading-6 uppercase"}>
+      <%= @resource_title %>
     </div>
     """
   end
 
   defp active_view_in_children?(children, active_view) do
-    Enum.any?(children, fn child -> child.view == active_view end)
+    Enum.any?(children, &(&1.view == active_view))
+  end
+
+  defp build_route(active_workspace, resource_slug, parent_view, view, sidebar_expanded) do
+    {_, _, [route]} =
+      case active_workspace do
+        :course_author ->
+          quote do:
+                  sigil_p(
+                    unquote(
+                      "/workspaces/course_author/#{resource_slug}/#{view}?sidebar_expanded=#{sidebar_expanded}"
+                    )
+                  )
+
+        :instructor ->
+          if parent_view do
+            quote do:
+                    sigil_p(
+                      unquote(
+                        "/workspaces/instructor/#{resource_slug}/#{parent_view}/#{view}?sidebar_expanded=#{sidebar_expanded}"
+                      )
+                    )
+          else
+            quote do:
+                    sigil_p(
+                      unquote(
+                        "/workspaces/instructor/#{resource_slug}/#{view}?sidebar_expanded=#{sidebar_expanded}"
+                      )
+                    )
+          end
+      end
+
+    route
```

Comment:
This technique avoids warnings coming from using sigil-p.

From ref: 

> You likely want this: path = sigil_p(unquote("/#{path}")) – build the static path before unquoting the result into the sigil macro call.

Ref: https://elixirforum.com/t/using-p-dynamically-inside-a-macro/56707/4